components/wifi/esp8266-driver/ESP8266Interface.cpp

```diff
@@ -511,14 +512,22 @@ int ESP8266Interface::socket_accept(void *server, void **socket, SocketAddress *
 
 int ESP8266Interface::socket_send(void *handle, const void *data, unsigned size)
 {
-    nsapi_error_t status;
+    nsapi_error_t status = NSAPI_ERROR_OK;
     struct esp8266_socket *socket = (struct esp8266_socket *)handle;
 
     if (!socket) {
         return NSAPI_ERROR_NO_SOCKET;
     }
 
-    status = _esp.send(socket->id, data, size);
+    unsigned long int sendStartTime = rtos::Kernel::get_ms_count();
+    while ((sendStartTime - rtos::Kernel::get_ms_count() < 100)
```

Comment:
This should be a `do while` loop - want to ensure we always call it at least once, no matter how much load is happening (and then you wouldn't need to initialise status above).